main/src/library/MutList.flix

```diff
@@ -56,27 +57,35 @@ instance Foldable[MutList] {
 
 namespace MutList {
 
+    ///
+    /// Constant which stores the minimum capacity of a MutList.
+    ///
+    def minCapacity(): Int32 = 8
+
     ///
     /// Returns a new empty mutable list with a default capacity.
     ///
-    pub def new(): MutList[a] & Impure = MutList(ref [], ref 0)
+    pub def new(): MutList[a] & Impure = MutList(ref [default; minCapacity()], ref 0)
 
     ///
     /// Returns a mutable list of all integers between `b` (inclusive) and `e` (exclusive).
     ///
     /// Returns an empty mutable list if `b >= e`.
     ///
-    pub def range(b: Int32, e: Int32): MutList[Int32] & Impure = MutList(ref Array.range(b, e), ref (e - b))
+    pub def range(b: Int32, e: Int32): MutList[Int32] & Impure =
+        let minCap = minCapacity();
+        if (e - b < minCap)
+            let f = i -> if (b + i < e) b + i else default;
+            MutList(ref Array.init(f, minCap), ref (e - b))
+        else
+            MutList(ref Array.range(b, e), ref (e - b))
 
     ///
     /// Returns the element at position `i` in the mutable list `v`.
     ///
     pub def get(i: Int32, v: MutList[a]): a & Impure =
         let MutList(a, _) = v;
-        if (i > length(v))
-            (deref a)[-1] /// deliberately trigger index-out-of-bounds
-        else
-            Array.get(deref a, i)
+        Array.get(deref a, i)
```

Comment:
I think this change is incorrect.

Imagine the array has 2 elements: a, b and capacity 4. Now if you ask for element 3 you should get a crash of some kind (perhaps better to use bug!), not whatever is at position 3.

Reply:
This still need to be fixed. It should check i vs. deref l and then use bug!